### PR TITLE
bpo-16355: fix a lack in inspect.getcomments() doc

### PR DESCRIPTION
**Pull request by marco-buttu**
Fix [bpo-16355](https://bugs.python.org/issue16355).

### CODE REVIEW

**Comment by berkerpeksag** [comment]
Note: Commit message should include "Initial patch by Vajrasky Kok.".

File: Lib/test/test_inspect.py [hunk 1]
```diff
@@ -24,8 +24,8 @@
 except ImportError:
     ThreadPoolExecutor = None
 
-from test.support import run_unittest, TESTFN, DirsOnSysPath, cpython_only
-from test.support import MISSING_C_DOCSTRINGS, cpython_only
+from test.support import (run_unittest, TESTFN, DirsOnSysPath, cpython_only,
+                          MISSING_C_DOCSTRINGS)
```
Comment (berkerpeksag) [inline Lib/test/test_inspect.py]: This doesn't look like a relevant change to bpo-16355. Please send a separate PR for this.

Anyway, I prefer hanging indentation style:

```py
from test.support import (
    MISSING_C_DOCSTRINGS, TESTFN, DirsOnSysPath, cpython_only, run_unittest,
)
```

File: Lib/test/test_inspect.py [hunk 2]
```diff
@@ -384,6 +384,12 @@ def test_cleandoc(self):
     def test_getcomments(self):
         self.assertEqual(inspect.getcomments(mod), '# line 1\n')
         self.assertEqual(inspect.getcomments(mod.StupidGit), '# line 20\n')
+        # If the object source file is not available, return None
+        fn = "_non_existing_filename_used_for_sourcefile_test.py"
```
Comment (berkerpeksag) [inline Lib/test/test_inspect.py]: `_non_existing_filename.py` is both short and descriptive. Also, this way you can avoid using a `fn` variable.

File: Lib/test/test_inspect.py [hunk 3]
```diff
@@ -384,6 +384,12 @@ def test_cleandoc(self):
     def test_getcomments(self):
         self.assertEqual(inspect.getcomments(mod), '# line 1\n')
         self.assertEqual(inspect.getcomments(mod.StupidGit), '# line 20\n')
+        # If the object source file is not available, return None
```
Comment (berkerpeksag) [inline Lib/test/test_inspect.py]: Finish sentences with full stops.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by marco-buttu** [comment]
Thanks @berkerpeksag, fixed

File: Lib/test/test_inspect.py [hunk 4]
```diff
@@ -384,6 +384,11 @@ def test_cleandoc(self):
     def test_getcomments(self):
         self.assertEqual(inspect.getcomments(mod), '# line 1\n')
         self.assertEqual(inspect.getcomments(mod.StupidGit), '# line 20\n')
+        # If the object source file is not available, return None.
+        co = compile("x=1", '_non_existing_filename.py', "exec")
```
Comment (berkerpeksag) [inline Lib/test/test_inspect.py]: Please use single quotes like the rest of the test.

**Comment by 1st1** [comment]
Why don't we raise an error if we can't find the source? Why returning `None`?

File: Lib/test/test_inspect.py (hunk 4 above)
Comment (marco-buttu) [reply Lib/test/test_inspect.py]: Thanks @berkerpeksag, done.

**Comment by marco-buttu** [comment]
@1st1, in [bpo-16355](http://bugs.python.org/issue16355) @ezio-melotti and @bitdancer discussed about raising an error or not, and eventually they agreed to return `None`.

**Comment by 1st1** [comment]
I see. The idea is to preserve the backwards compatibility. Then LGTM.

File: Doc/library/inspect.rst [hunk 5]
```diff
@@ -442,7 +442,9 @@ Retrieving source code
 
    Return in a single string any lines of comments immediately preceding the
    object's source code (for a class, function, or method), or at the top of the
-   Python source file (if the object is a module).
+   Python source file (if the object is a module).  If the object's source code
+   is unavailable, return ``None``.  This could happen if the object has been
+   defined in C or interactive shell.
```
Comment (vadmium) [inline Doc/library/inspect.rst]: or the interactive shell

**Review by berkerpeksag — APPROVED** [review]

**Comment by berkerpeksag** [comment]
Thanks!

**Pull request merged**